src/Orchard.Web/Modules/Orchard.ImportExport/Services/SetupService.cs

```diff
@@ -101,6 +102,10 @@ public class SetupService : Component, ISetupService {
                     schemaBuilder.AlterTable("Orchard_Framework_DataMigrationRecord",
                         table => table.AddUniqueConstraint("UC_DMR_DataMigrationClass_Version", "DataMigrationClass", "Version"));
```

Comment:
It's necessary for the Import screen where the user has the option to blow away the entire database and setup the site using the uploaded recipe file. The fundamental difference between this SetupService and the one from Orchard.Setup is the way the recipe is provided I believe, so it should in principle not be too difficult to refactor some bits and bobs. I'll add a card for it.

I'm not 100% sure what you mean with your second part of your comment. Do you mean we should have a single `CoreSchemaBuilder` class that handles both the creation of the DataMigrationRecord table as well as the DistributedLockRecord table (including related key constraints)?